### PR TITLE
Add the --print-extras-to(-file) option

### PR DESCRIPTION
**Pull request by hroncok**
TODO:

- [x] Add more tests
- [x] Add documentation
- [x] Support print deps to file and print extras to file as one action

### CODE REVIEW

**Comment by hroncok** [comment]
@encukou Could you please review this as a proof of concept?

**Comment by encukou** [comment]
The idea sounds good, but I don't think `--print-extras-only` is useful for anything.
I can do a complete review tomorrow.

**Comment by hroncok** [comment]
`--print-extras-only` is created to mirror `--print-deps-only`

If we handle https://github.com/fedora-python/tox-current-env/issues/18 first, we can get rid of it, otherwise, I'd rather keep the options in sync.

**Comment by hroncok** [comment]
The print deps tests were failing for me on some of the toxenvs and I've tried everything. At the end, it turned out to be a race condition with clogged files. Added flushing and it works.

I'll add documentation later, tests + implementation ready fro review.

File: README.rst [hunk 1]
```diff
@@ -5,7 +5,7 @@ tox-current-env
 `tox <https://tox.readthedocs.io/>`_  plugin to run tests in current Python environment
 ---------------------------------------------------------------------------------------
 
-The ``tox-current-env`` plugin adds two options:
+The ``tox-current-env`` plugin adds three options:
```
Comment (encukou) [inline README.rst:8]: Minor suggestion, I usually use troll-style counting (one, two, many) when maintaining lists:
```suggestion
The ``tox-current-env`` plugin adds these options:
```

File: src/tox_current_env/hooks.py [hunk 2]
```diff
@@ -38,6 +38,16 @@ def tox_addoption(parser):
         help="Don't run tests, only print the dependencies to the given file "
             + "(use `-` for stdout)",
     )
+    parser.add_argument(
+        "--print-extras-to",
+        "--print-extras-to-file",
```
Comment (encukou) [inline src/tox_current_env/hooks.py:43]: Consider removing `--print-extras-to-file` – there's no backwards compatibility here to keep.

File: src/tox_current_env/hooks.py [hunk 3]
```diff
@@ -55,11 +65,17 @@ def tox_configure(config):
...
-    if config.option.current_env or config.option.print_deps_to:
+    if config.option.current_env or config.option.print_deps_to or config.option.print_extras_to:
         config.skipsdist = True
         for testenv in config.envconfigs:
             config.envconfigs[testenv].whitelist_externals = "*"
 
+    if (getattr(config.option.print_deps_to, "name", object()) ==
+        getattr(config.option.print_extras_to, "name", object())):
+            raise tox.exception.ConfigError(
+                "The paths given to --print-deps-to and --print-extras-to cannot be identical."
```
Comment (encukou) [inline src/tox_current_env/hooks.py:76]: Whoa I wouldn't have thought of that edge case!

**Review by encukou — APPROVED** [review]
Good to merge, with just “many” tiny comments/suggestions:

File: src/tox_current_env/hooks.py (hunk 2 above)
Comment (hroncok) [reply src/tox_current_env/hooks.py:43]: I'd keep it, it's not a maintenance burden and it makes things nicely similar.

**Pull request merged**